lib/browser/api/menu.js

```diff
@@ -5,181 +5,77 @@ const EventEmitter = require('events').EventEmitter
-      return command != null ? command.icon : undefined
-    },
-    executeCommand: (event, commandId) => {
-      const command = this.commandsMap[commandId]
-      if (command == null) return
+    getIconForCommandId: id => this.commandsMap[id].icon,
+    executeCommand: (event, id) => {
+      const command = this.commandsMap[id]
+      if (command === null) return
       command.click(event, BrowserWindow.getFocusedWindow(), webContents.getFocusedWebContents())
     },
     menuWillShow: () => {
-      // Make sure radio groups have at least one menu item seleted.
-      var checked, group, id, j, len, radioItem, ref1
-      ref1 = this.groupsMap
-      for (id in ref1) {
-        group = ref1[id]
-        checked = false
-        for (j = 0, len = group.length; j < len; j++) {
-          radioItem = group[j]
-          if (!radioItem.checked) {
-            continue
-          }
-          checked = true
-          break
-        }
-        if (!checked) {
-          v8Util.setHiddenValue(group[0], 'checked', true)
-        }
+      // Ensure radio groups have at least one menu item seleted
+      for (let id in this.groupsMap) {
+        let found = this.groupsMap[id].find(item => item.checked) || null
+        if (!found) v8Util.setHiddenValue(this.groupsMap[id][0], 'checked', true)
       }
     }
   }
 }
 
+// create and show a popup
 Menu.prototype.popup = function (window, x, y, positioningItem) {
   let asyncPopup
+  let [newX, newY, newPosition, newWindow] = [x, y, positioningItem, window]
 
   // menu.popup(x, y, positioningItem)
-  if (window != null && (typeof window !== 'object' || window.constructor !== BrowserWindow)) {
-    // Shift.
-    positioningItem = y
-    y = x
-    x = window
-    window = null
+  if (window !== null) {
+    // shift over values
+    if (typeof window !== 'object' || window.constructor !== BrowserWindow) {
+      [newPosition, newY, newX, newWindow] = [y, x, window, null]
+    }
   }
 
   // menu.popup(window, {})
-  if (x != null && typeof x === 'object') {
-    const options = x
-    x = options.x
-    y = options.y
-    positioningItem = options.positioningItem
-    asyncPopup = options.async
+  if (x !== null && typeof x === 'object') {
+    const opts = x
+    newX = opts.x
+    newY = opts.y
+    newPosition = opts.positioningItem
+    asyncPopup = opts.async
   }
 
-  // Default to showing in focused window.
-  if (window == null) window = BrowserWindow.getFocusedWindow()
-
-  // Default to showing under mouse location.
-  if (typeof x !== 'number') x = -1
-  if (typeof y !== 'number') y = -1
-
-  // Default to not highlighting any item.
-  if (typeof positioningItem !== 'number') positioningItem = -1
-
-  // Default to synchronous for backwards compatibility.
+  // set defaults
+  if (typeof x !== 'number') newX = -1
+  if (typeof y !== 'number') newY = -1
+  if (typeof positioningItem !== 'number') newPosition = -1
+  if (window === null) newWindow = BrowserWindow.getFocusedWindow()
```

Comment:
Why not just `!window`? Are there cases in which the window is falsy but you do want it?